Wikipedia/Code/UINavigationBar+WMFImageGalleryGradient.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import QuartzCore
+
+extension UINavigationBar {
+    
+    func wmf_applyGalleryTopGradientBackground() {
+        if (!CGRectIsNull(bounds)){
+            if (CGRectGetWidth(bounds) > 0 && CGRectGetHeight(bounds) > 0){
+                setBackgroundImage(wmf_galleryTopGradientBackgroundImage(), forBarMetrics: .Default)
+            }
+        }
+    }
+    
+    func wmf_galleryTopGradientBackgroundImage() -> UIImage {
```

Comment:
Looking here it seems you do not have to render to an image. This library uses CAGradientLayer to achieve the same effect:

https://github.com/chroman/CRGradientNavigationBar/blob/master/CRGradientNavigationBar/CRGradientNavigationBar.m